packages/components/pie-card-container/src/index.ts

```diff
@@ -1,16 +1,63 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-
+import {
+    html, LitElement, unsafeCSS, nothing,
+} from 'lit';
+import { property } from 'lit/decorators.js';
+import { validPropertyValues } from '@justeattakeaway/pie-webc-core';
 import styles from './card-container.scss?inline';
-import { CardContainerProps } from './defs';
+import { CardContainerProps, type AriaProps, variants } from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-card-container';
 
 export class PieCardContainer extends LitElement implements CardContainerProps {
+    @property()
+    @validPropertyValues(componentSelector, variants, 'default')
+    public variant: CardContainerProps['variant'] = 'default';
+
+    @property({ type: String, reflect: true })
+    public href?: string;
+
+    @property({ type: String, reflect: true })
+    public target?: string;
+
+    @property({ type: String, reflect: true })
+    public rel?: string;
+
+    @property({ type: Boolean })
+    public disabled = false;
+
+    @property({ type: Object })
+    public aria!: AriaProps;
+
     render () {
-        return html`<h1 data-test-id="pie-card-container">Hello world!</h1>`;
+        const {
+            href,
+            target,
+            rel,
+            disabled,
+            variant,
+        } = this;
+
+        return html`
+                <div
+                    variant=${variant}
+                    class="c-card-container"
+                    data-test-id="pie-card-container"
+                    ?disabled=${disabled}>
+                    <a
```

Comment:
should this be conditional only if the href is available?
maybe we can consider using a similar approach to the link component?

Reply:
I think for a starting point this is ok - the original ticket had no mention of link behaviour (but it is required to add interactive states) so I would prefer not to add too much logic and instead refine the desired behaviours with Kevin and create more tickets.

This is just a very, very basic starting point and can be improved on as we scope out the required behaviours